index.js

```diff
@@ -2,16 +2,24 @@ var mongoose = require('mongoose');
 
 module.exports = function(schema) {
   var pathsToPopulate = [];
-  schema.eachPath(function (pathname, schemaType) {
-    if (schemaType.options && schemaType.options.autopopulate) {
-      var option = schemaType.options.autopopulate;
-      pathsToPopulate.push({ path: pathname, autopopulate: option });
-    }
-  });
+  (function recurseSchema(schema, path) {
```

Comment:
I don't use 'use strict' in this module, but I really should. This function should not be in the module.exports function's scope, especially since it doesn't use any variables outside of its parameters.